src/types.ts

```diff
@@ -106,6 +106,13 @@ export interface ElectronDownloadRequestOptions {
    * It is used before artifacts are put into cache.
    */
   tempDirectory?: string;
+  /**
+   * When set to `true`, do not put the downloaded artifact into the cache, leaving it
+   * in the temporary directory. You need to clean up the temporary directory yourself.
+   *
+   * Defaults to `false`.
+   */
+  dontCache?: boolean;
```

Comment:
```suggestion
  disable?: boolean;
```

naming negations can be annoying, but we already have `unsafelyDisableChecksums` so maybe this should be `disableCache` to follow suit.

Reply:
Unfortunately the naming here is going to be annoying to nail down. The option doesn't disable *reading* from the cache, only *writing* to it. So I think `disableCache` isn't quite right either since it implies not using the cache at all.

The `force` option skips *reading* from the cache but does *write* to the cache.

You can see my deliberations with myself about the name from 2.5 years ago in #290. 😅

Reply:
Basically I think we want to support four states and name this option such that it logically makes sense with them:

* Read and Write to Cache - Default behavior with no options specified
* Only Read from Cache - `<this-option>: true`
* Only Write to Cache - `force: true`
* No Cache at All - `force: true` and `<this-option>: true`

Reply:
Or we have the default be true and name the option `saveToCache`, and you set it to false to disable it.

Reply:
Don't want to spend too much time bikeshedding on this, but `saveToCache: true` sounds good to me since I think `dontCache` can be ambiguous.

Reply:
I love bikeshedding, I propose we deprecaate `force: true` and introduce a new option.

`cacheMode: CacheMode`

```typescript
enum CacheMode {
  // This is the default, reads from cache if it's there,
  // writes to cache and return value at the same time if not
  ReadWrite,
  // Reads the value from the cache but will not write to the
  // cache if it has to fetch
  ReadOnly,
  // Skips reading from the cache but will persist to the cache
  // and overwrite whatever is in there, this is what `force: true`
  // should map to
  WriteOnly,
  // Bypasses the cache completely and neither reads nor writes
  Bypass,
}
```

Reply:
Then we can make all shasum fetching `cacheMode: Bypass`. We should note in the docs that in the `Bypass` mode it is the callers responsibility to delete the file once you're done with it.

I'm also not convinced there is a usecase for `ReadOnly` so maybe we can skip that

Reply:
> I'm also not convinced there is a usecase for `ReadOnly` so maybe we can skip that

`fiddle-core` would use that mode if it existed. The vast majority of folks wouldn't ever get a cache hit, but for those of us doing other development using `@electron/get` which populates the cache, `fiddle-core` could just copy out the version that already exists on your disk rather than downloading it again. It's a minor optimization but still an optimization. 🤷

Otherwise `fiddle-core` could just use `Bypass` mode.

Reply:
Yeah that's what I wanted to avoid 🤔 A `ReadOnly` mode _sometimes_ leaves the file in the ownership of the caller, and sometimes doesn't.

Maybe in the `ReadOnly` mode if it's in the cache we copy it to the tmp directory anyway and then it's always owned by the caller?

Reply:
In broad strokes, I like the enum idea because it's cleaner than needing to configure two separate options and making sure they match up (echoes of `webPreferences` 😱 ).

Reply:
> Maybe in the `ReadOnly` mode if it's in the cache we copy it to the tmp directory anyway and then it's always owned by the caller?

That's what this PR already does.

https://github.com/electron/get/blob/5df6263ce4b386ff6632a96af9d3b5fa89fe51f3/src/types.ts#L109-L114

https://github.com/electron/get/blob/5df6263ce4b386ff6632a96af9d3b5fa89fe51f3/src/index.ts#L156-L161